src/granite_orm.cr

```diff
@@ -48,6 +48,31 @@ class Granite::ORM
     {% SETTINGS[:timestamps] = true %}
   end
 
+  # retrive the parent relationship
+  macro belongs_to(model_name)
+    {% FIELDS["#{model_name.id}_id".id.symbolize] = Int64 %}
+
+    def {{model_name.id}}
+      parent = {{model_name.id.camelcase}}.find {{model_name.id}}_id
+      return {{model_name.id.camelcase}}.new unless parent
+      parent
```

Comment:
I prefer
```cr
if parent = {{model_name.id.camelcase}}.find {{model_name.id}}_id
  parent
else
  {{model_name.id.camelcase}}.new
end
```

Reply:
Can you provide a setter as well?
```
user.post = post
```
This should set the `post_id` to `post.id`

Reply:
@c910335 I will change it to your preferred format
@drujensen Will do!